pkgs/development/python-modules/jax-cuda12-plugin/default.nix

```diff
@@ -94,12 +95,34 @@ buildPythonPackage {
     wheelUnpackHook
   ];
 
+  # jax-cuda12-plugin looks for ptxas at runtime, e.g. with a triton kernel.
+  # Linking into $out is the least bad solution. See
+  # * https://github.com/NixOS/nixpkgs/pull/164176#discussion_r828801621
+  # * https://github.com/NixOS/nixpkgs/pull/288829#discussion_r1493852211
+  # * https://github.com/NixOS/nixpkgs/pull/375186
+  # for more info.
+  postInstall = ''
+    mkdir -p $out/${python.sitePackages}/jax_cuda12_plugin/cuda/bin
+    ln -s ${lib.getExe' cudaPackages.cuda_nvcc "ptxas"} $out/${python.sitePackages}/jax_cuda12_plugin/cuda/bin
+    ln -s ${lib.getExe' cudaPackages.cuda_nvcc "nvlink"} $out/${python.sitePackages}/jax_cuda12_plugin/cuda/bin
+  '';
+
+  # jax-cuda12-plugin contains shared libraries that open other shared libraries via dlopen
+  # and these implicit dependencies are not recognized by ldd or
+  # autoPatchelfHook. That means we need to sneak them into rpath. This step
+  # must be done after autoPatchelfHook and the automatic stripping of
+  # artifacts. autoPatchelfHook runs in postFixup and auto-stripping runs in the
+  # patchPhase.
```

Comment:
IIRC we used to handle this situation with `patchelf --add-needed` before `autoPatchelfHook`. That way we could get rid of the FIXME below. @SomeoneSerge would prob know better than I?

Reply:
I copied this verbatim from `jax-cuda12-pjrt` (https://github.com/NixOS/nixpkgs/pull/369920); I think this can be cleaned up in both later.

https://github.com/NixOS/nixpkgs/blob/f9c16aa4768850e24021d1ae684125177c8ebfe3/pkgs/development/python-modules/jax-cuda12-pjrt/default.nix#L77-L85

Reply:
roger that, sounds good